### PR TITLE
add a few no-file tests

### PR DESCRIPTION
**Pull request by onyxraven**
Background
-----

Adds a few 'no gitignore' tests. Followup of #50 

Also prefers to add a newline at the end of the gitignore.

Versioning
-----

batched up in the next release

Tasks
-----

* [x] Specs written
* [x] Manual testing

/cc
-----
@randrusiak

### CODE REVIEW

File: scm/git.go [hunk 1]
```diff
@@ -121,9 +115,10 @@ func appendLineToFileIfNotExists(line string, filename string) error {
 		}
 	}
 
-	if _, err := file.WriteString(line); err != nil {
+	if _, err := file.WriteString(line + "\n"); err != nil {
```
Comment (onyxraven) [inline scm/git.go:116]: prefers adding the newline.

File: scm/git_test.go [hunk 2]
```diff
@@ -48,17 +48,39 @@ func TestRemoveFileFromIgnored(t *testing.T) {
 	}
 }
 
+func TestRemoveFileNoExistingFile(t *testing.T) {
+	t.Run("missing gitignore file", func(t *testing.T) {
+		tempFilePath := t.TempDir() + "/.gitignore"
+		if _, err := os.Stat(tempFilePath); err == nil {
+			t.Errorf("RemoveFileFromIgnored() = gitignore accidentally exists")
```
Comment (onyxraven) [inline scm/git_test.go:55]: specifically test that we're not using some leftover from another test

File: scm/git_test.go [hunk 3]
```diff
@@ -48,23 +48,45 @@ func TestRemoveFileFromIgnored(t *testing.T) {
...
+		git := Git{
+			IgnoreFilePath: tempFilePath,
+		}
+		err := git.RemoveFileFromIgnored("secret.yaml")
+		if err != nil {
+			t.Errorf("RemoveFileFromIgnored() err=%v", err)
+		}
+
+		if _, err := os.Stat(tempFilePath); err == nil {
+			t.Errorf("RemoveFileFromIgnored() = gitignore exists when it should not")
```
Comment (onyxraven) [inline scm/git_test.go:67]: ensure we didn't accidentally do something to the file (which still shouldn't exist)

File: scm/git_test.go [hunk 4]
```diff
@@ -48,23 +48,45 @@ func TestRemoveFileFromIgnored(t *testing.T) {
...
+		{name: "Add file to end", gitIgnoreContent: []byte("test.txt\nvendor"), fileName: "secret.yml", want: []byte("test.txt\nvendor\nsecret.yml\n")},
+		{name: "Add file to end, trailing newline", gitIgnoreContent: []byte("test.txt\nvendor\n"), fileName: "secret.yml", want: []byte("test.txt\nvendor\nsecret.yml\n")},
+		{name: "Add file with path", gitIgnoreContent: []byte("test.txt\nvendor"), fileName: "../../../secret.yml", want: []byte("test.txt\nvendor\n../../../secret.yml\n")},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			tempFilePath := t.TempDir() + "/.gitignore"
-			err := os.WriteFile(tempFilePath, tt.gitIgnoreContent, 0644)
+			err := os.WriteFile(tempFilePath, tt.gitIgnoreContent, 0600)
```
Comment (onyxraven) [inline scm/git_test.go:89]: linter wanted this, and in a test, this makes sense

**Review by dennis-johnson-dev — APPROVED** [review]
🚢

File: scm/git_test.go [hunk 5]
```diff
@@ -90,3 +112,30 @@ func TestAddFileToIgnored(t *testing.T) {
 		})
 	}
 }
+
+func TestAddFileNoExistingFile(t *testing.T) {
```
Comment (dennis-johnson-dev) [inline scm/git_test.go:116]: Thanks for the test 👍

**Review by elementalvoid — APPROVED** [review]

**Pull request merged**